### PR TITLE
20210411 domain throttling

### PR DESCRIPTION
**Pull request by peterthomassen**

### CODE REVIEW

File: api/desecapi/views.py [hunk 1]
```diff
@@ -70,14 +70,19 @@ class DomainViewMixin:
...
         # noinspection PyUnresolvedReferences
-        super().initial(request, *args, **kwargs)
+        return self.request.user.domains.get(name=self.kwargs['name'])
+
+    def check_throttles(self, request):
         try:
-            # noinspection PyAttributeOutsideInit, PyUnresolvedReferences
-            self.domain = self.request.user.domains.get(name=self.kwargs['name'])
+            # noinspection PyStatementEffect
+            self.domain  # call the method behind the property
```
Comment (nils-wisiol) [inline api/desecapi/views.py:93]: that's a weird usage of a property. Maybe convert it to a method?

Comment (peterthomassen) [reply api/desecapi/views.py]: right. Did it differently now

File: api/desecapi/throttling.py [hunk 2]
```diff
@@ -25,6 +25,11 @@ def allow_request(self, request, view):
         if self.rate is None:
             return True
 
+        # Amend scope with optional bucket
+        bucket = getattr(view, self.scope_attr + '_bucket', None)
+        if bucket is not None:
+            self.scope += ':' + bucket
```
Comment (nils-wisiol) [inline api/desecapi/throttling.py:31]: why not let the view decide that

**Review by nils-wisiol — APPROVED** [review]
lgtm

**Pull request merged**